FirebaseSessions/Sources/FirebaseSessions.swift

```diff
@@ -0,0 +1,55 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import Foundation
+import FirebaseCore
+
+// Avoids exposing internal FirebaseCore APIs to Swift users.
+@_implementationOnly import FirebaseCoreExtension
+
+@objc(FIRSessionsProvider)
+protocol SessionsProvider {
+  @objc func sessions(for app: FirebaseApp) -> Sessions
+}
+
+@objc(FIRSessions) class Sessions: NSObject, Library, SessionsProvider {
+  // MARK: - Private Variables
+
+  /// The app associated with all sessions.
+  private let app: FirebaseApp
+
+  // MARK: - Initializers
+
+  required init(app: FirebaseApp) {
+    self.app = app
+  }
+
+  // MARK: - Library conformance
+
+  static func componentsToRegister() -> [Component] {
+    return [Component(SessionsProvider.self,
+                      instantiationTiming: .eagerInDefaultApp,
```

Comment:
That's a good question - so I had assumed it would try and start it up again for the 2nd app if it was alwaysEager - is that the case, or is that controlled by something else? We only want the Sessions SDK to startup for the default app (like Crashlytics and Performance do).

Reply:
I spoke with Ryan about this and I think since the long term design is to make this available to other SDKs, we could use `.alwaysEager` and then, for now, check in the block whether to instantiate based on if it's a default app. See below suggestion.